packages/app/src/settings/device/NotificationSettings.vue

```diff
@@ -0,0 +1,154 @@
+        </Button>
+        <i18n-t
+          tag="p"
+          scope="global"
+          keypath="settingsPage.notifications.notReceivingNotifications"
+          class="pt-[15px] text-gray-600 text-sm font-normal"
+        >
+          <ExternalLink href="#">
+            {{ t('settingsPage.notifications.troubleshoot') }}
+          </ExternalLink>
+        </i18n-t>
+      </div>
+    </div>
+  </SettingsSection>
+</template>
+
+<script lang="ts" setup>
+// TODO: Add documentation link https://github.com/cypress-io/cypress-documentation/issues/5280
+import { ref, watch } from 'vue'
+import { gql, useMutation } from '@urql/vue'
+import { useI18n } from '@cy/i18n'
+import Switch from '@packages/frontend-shared/src/components/Switch.vue'
+import SettingsSection from '../SettingsSection.vue'
+import { NotificationSettingsFragment, SetNotificationSettingsDocument, NotificationSettings_ShowNotificationDocument } from '../../generated/graphql'
+import Checkbox from '@cy/components/Checkbox.vue'
+import Button from '@cypress-design/vue-button'
+import ExternalLink from '@cy/gql-components/ExternalLink.vue'
+
+const { t } = useI18n()
+
+const props = defineProps<{
+  gql: NotificationSettingsFragment
+}>()
+
+gql`
+fragment NotificationSettings on Query {
+  localSettings {
+    preferences {
+      notifyWhenRunStarts
+      notifyWhenRunStartsFailing
+      notifyWhenRunCompletes
+    }
+  }
+}
+`
+
+gql`
+mutation SetNotificationSettings($value: String!, $merge: Boolean) {
+  setPreferences (merge: $merge, value: $value, type: global) {
+    ...NotificationSettings
+  }
+}`
+
+gql`
+mutation NotificationSettings_ShowNotification($title: String!, $body: String!) {
+  showSystemNotification(title: $title, body: $body)
+}`
+
+const switches = [
+  {
+    id: 'notifyWhenRunStarts',
+    title: t('settingsPage.notifications.notifyMeWhenRunStarts'),
+  },
+  {
+    id: 'notifyWhenRunStartsFailing',
+    title: t('settingsPage.notifications.notifyMeWhenRunIsFailing'),
+  },
+]
+
+const listRef = ref(props.gql.localSettings.preferences.notifyWhenRunCompletes)
+
+const statuses = [
+  { id: 'passed', label: t('settingsPage.notifications.passed') },
+  { id: 'failed', label: t('settingsPage.notifications.failed') },
+  { id: 'canceled', label: t('settingsPage.notifications.canceled') },
+  { id: 'errored', label: t('settingsPage.notifications.errored') },
+]
+
+const setPreferences = useMutation(SetNotificationSettingsDocument)
+const showNotification = useMutation(NotificationSettings_ShowNotificationDocument)
+
+async function showTestNotification () {
+  await showNotification.executeMutation({ title: t('settingsPage.notifications.testNotificationTitle'), body: t('settingsPage.notifications.testNotificationBody') })
+}
+
+function updatePref (value: boolean, property: string) {
+  setPreferences.executeMutation({
+    value: JSON.stringify({ [property]: value }),
+    merge: true,
+  })
+}
+
+watch(() => listRef.value, async (newVal) => {
```

Comment:
Maybe a good use case for [`watchDebounced`](https://vueuse.org/shared/watchDebounced/) to minimize the number of writes

Reply:
Good idea, added